source/code/plugin/filter_cadvisor2mdm.rb

```diff
@@ -145,7 +145,12 @@ def ensure_cpu_memory_capacity_set
             end
 
             begin
-                nodeInventory = JSON.parse(KubernetesApiClient.getKubeResourceInfo("nodes?fieldSelector=metadata.name%3D#{@@hostName}").body)
+                resourceUri = "nodes?fieldSelector=metadata.name%3D#{@@hostName}"
+                # For ARO, filter out all other node roles other than compute
+                if KubernetesApiClient.isAROCluster
+                    resourceUri = resourceUri + "&labelSelector=node-role.kubernetes.io/compute%3Dtrue"
```

Comment:
so when u pass in fieldselector & label selector, is it an AND between those two?

Reply:
> so when u pass in fieldselector & label selector, is it an AND between those two?

yes, that's right.